src/js/Mixins/Snapping.js

```diff
@@ -149,10 +150,19 @@ const SnapMixin = {
         const distanceBC = this._getDistance(map, B, C);
 
         // closest latlng of A and B to C
-        const closestVertexLatLng = distanceAC < distanceBC ? A : B;
+        let closestVertexLatLng = distanceAC < distanceBC ? A : B;
 
         // distance between closestVertexLatLng and C
-        const shortestDistance = distanceAC < distanceBC ? distanceAC : distanceBC;
+        let shortestDistance = distanceAC < distanceBC ? distanceAC : distanceBC;
+
+        // snap to middle of segment if enabled
+        if (this.options.snapMiddle) {
+            const M = this._middleLatLng(map, A, B);
+            const distanceMC = this._getDistance(map, M, C);
+            const middleIsNearest = (distanceMC < distanceAC && distanceMC < distanceBC);
+            closestVertexLatLng = middleIsNearest ? M : closestVertexLatLng;
+            shortestDistance = middleIsNearest ? distanceMC : shortestDistance;
+        }
```

Comment:
should we add a few more explanation commends between those lines to explain what exactly is done inside the if-clause?